go/engine/passphrase_change.go

```diff
@@ -149,66 +155,48 @@ func (c *PassphraseChange) forceUpdatePassphrase(m libkb.MetaContext, sigKey lib
 		m.CDebugf("PassphraseChange.runForcedUpdate: Losing %v synced keys", nPgpKeysLost)
 	}
 
-	var acctErr error
-	c.G().LoginState().Account(func(a *libkb.Account) {
-		// Ready the update argument; almost done, but we need some more stuff.
-		m = m.WithLoginContext(a)
-		payload, err := c.commonArgs(a, oldClientHalf, pgpKeys, ppGen)
-		if err != nil {
-			acctErr = err
-			return
-		}
-
-		// get the new passphrase hash out of the args
-		pwh, ok := payload["pwh"].(string)
-		if !ok || len(pwh) == 0 {
-			acctErr = errors.New("no pwh found in common args")
-			return
-		}
+	// Ready the update argument; almost done, but we need some more stuff.
+	payload, err := c.commonArgs(m, oldClientHalf, pgpKeys, ppGen)
+	if err != nil {
+		return err
+	}
 
-		// get the new PDPKA5 KID out of the args
-		pdpka5kid, ok := payload["pdpka5_kid"].(string)
-		if !ok || len(pdpka5kid) == 0 {
-			acctErr = errors.New("no pdpka5kid found in common args")
-			return
-		}
+	// get the new passphrase hash out of the args
+	pwh, ok := payload["pwh"].(string)
+	if !ok || len(pwh) == 0 {
+		return errors.New("no pwh found in common args")
+	}
 
-		// Generate a signature with our unlocked sibling key from device.
-		proof, err := c.me.UpdatePassphraseProof(sigKey, pwh, ppGen+1, pdpka5kid)
-		if err != nil {
-			acctErr = err
-			return
-		}
+	// get the new PDPKA5 KID out of the args
+	pdpka5kid, ok := payload["pdpka5_kid"].(string)
+	if !ok || len(pdpka5kid) == 0 {
+		return errors.New("no pdpka5kid found in common args")
+	}
 
-		sig, _, _, err := libkb.SignJSON(proof, sigKey)
-		if err != nil {
-			acctErr = err
-			return
-		}
-		payload["sig"] = sig
-		payload["signing_kid"] = sigKey.GetKID()
-
-		postArg := libkb.APIArg{
-			Endpoint:    "passphrase/sign",
-			SessionType: libkb.APISessionTypeREQUIRED,
-			JSONPayload: payload,
-			SessionR:    a.LocalSession(),
-		}
+	// Generate a signature with our unlocked sibling key from device.
+	proof, err := c.me.UpdatePassphraseProof(sigKey, pwh, ppGen+1, pdpka5kid)
+	if err != nil {
+		return err
+	}
 
-		_, err = c.G().API.PostJSON(postArg)
-		if err != nil {
-			acctErr = fmt.Errorf("api post to passphrase/sign error: %s", err)
-			return
-		}
+	sig, _, _, err := libkb.SignJSON(proof, sigKey)
+	if err != nil {
+		return err
+	}
+	payload["sig"] = sig
+	payload["signing_kid"] = sigKey.GetKID()
 
-		// Reset passphrase stream cache so that subsequent updates go through
-		// without a problem (see CORE-3933)
-		a.ClearStreamCache()
-	}, "PassphraseChange.runForcedUpdate")
-	if acctErr != nil {
-		return acctErr
+	postArg := libkb.APIArg{
+		Endpoint:    "passphrase/sign",
+		SessionType: libkb.APISessionTypeREQUIRED,
```

Comment:
Without SessionR, doesn't this end up going through LoginState to get the session?

Reply:
Not if you pass a `MetaContext` through, which will have a provisional login context.  I'll comment to that effect.  Eventually, I'd like to retire `SessionR`.

Reply:
ok, there's another libkb.APIArg in this PR where you added MetaContext but left SessionR.